apps/site/pages/teaching/courses/create.tsx

```diff
@@ -6,13 +6,32 @@ import { useRouter } from "next/router";
 
 export default function CreateCoursePage() {
 	const { mutateAsync: createCourse } = trpc.course.create.useMutation();
+	const { mutateAsync: addCourse } = trpc.specialization.addCourse.useMutation();
 	const router = useRouter();
+	const { specializationId } = router.query;
+	const { subjectId } = router.query;
 	const session = useRequiredSession();
 	const author = session.data?.user.name;
 
 	async function onConfirm(course: CourseFormModel) {
 		try {
-			const { title, slug } = await createCourse(course);
+			const { title, slug, courseId } = await createCourse(course);
+
+			if (
+				specializationId !== null &&
+				specializationId !== undefined &&
+				typeof specializationId === "string" &&
+				subjectId !== null &&
+				subjectId !== undefined &&
+				typeof subjectId === "string"
+			) {
```

Comment:
zwar eine gute idee, aber die validierung haben wir bisher einfache durchgeführt
simpler cast innerhalb eines try-catch. Da du an der stelle in einem block bist, kannst du einfach `subjectId: subjectId as string` nutzen. Falls das in runtime nicht möglich ist, würde ein Fehler geworfen werden un der nutzer kriegt den toast angezeigt.
folgendes würde ausreichen:
```ts
			await addCourse({
					subjectId: subjectId as string,
					specializationId: specializationId as string,
					courseId: courseId as string
				});

```
Im ganz konkreten fall ist aber nicht einmal der manuelle cast notwendig da zod mit trpc das automatisch übernimmt und einen fehler wirft. (zod ist hier für die validierung zuständig). also reicht folgendes aus innerhalb des try-catch aus ohne vorherige checks: 
```ts
			await addCourse({
					subjectId: subjectId,
					specializationId: specializationId,
					courseId: courseId
				});
```

Reply:
```Typescript
await addCourse({
	subjectId: subjectId as string,
	specializationId: specializationId as string,
	courseId: courseId as string
});
```
Es klappt nur die erste Version, weil der Compiler  denke die sachen könnten null oder undefined sein, Habe ich so übernommen. Ich dachte "as" casts sollen wir vermeinden oder ging es da nur um "as any"?